docs/faq.md

```diff
@@ -115,9 +118,92 @@ in their default build configuration due to slow performance of the MacOS sandbo
 stanza adds it back. [An issue](https://github.com/cgrindel/rules_swift_package_manager/issues/712)
 exists tracking the work to allow these Swift packages to be built using the `local` spawn strategy.
 
+## How do I handle the error `Unable to resolve byName reference XXX in @swiftpkg_yyy.`?
+
+tl;dr A transitive dependency uses a by-name product reference format that was deprecated in Swift
+5.2. You need to patch the Swift package to use a more explicit product reference.
+
+### How do I fix this issue?
+
+[Patch the Swift package] to use a fully-qualified product reference.
+
+Let's look at an example. At the time of this writing, the [pusher-websocket-swift Package.swift]
+refers to the `TweetNacl` product in the `tweetnacl-swiftwrap` package using just the name of the
+product.
+
+```swift
+// swift-tools-version:5.0
+
+import PackageDescription
+
+let package = Package(
```

Comment:
I'd probably just leave the relevant bits like the target, name, and package version to reduce noise